import_export_celery/tasks.py

```diff
@@ -188,44 +188,58 @@ def run_import_job(pk, dry_run=True):
         return
 
 
-@task(bind=False)
+@task(bind=False,soft_time_limit=36000)
 def run_export_job(pk):
     log.info("Exporting %s" % pk)
     export_job = models.ExportJob.objects.get(pk=pk)
+    pks = json.loads(export_job.queryset)
+    qs_len = len(pks)
     resource_class = export_job.get_resource_class()
-    queryset = export_job.get_queryset()
-    qs_len = len(queryset)
+    pk_list =[],
+    rows = 0
+    interval = 100000
+
+    def TempFileSave(filename, content):
+        with open(filename, "ab") as myfile:
+            myfile.write(content)
+
+    def FileSave(filename):
+        reopen = open(filename, 'rb')
+        django_file = File(reopen)
+        export_job.file.save(filename, django_file,save= True)
 
     class Resource(resource_class):
         def __init__(self, *args, **kwargs):
             self.row_number = 1
             super().__init__(*args, **kwargs)
 
         def export_resource(self, *args, **kwargs):
-            if self.row_number % 20 == 0 or self.row_number == 1:
-                change_job_status(
-                    export_job,
-                    "export",
-                    "Exporting row %s/%s" % (self.row_number, qs_len),
-                )
+            if self.row_number%2000 == 0 or self.row_number == 1:
```

Comment:
`2000` is way to high. Do you have any evidence that such a high value is needed? Does this need to be customizable using django SETTINGS?

Reply:
For millions of rows. 20 was very small. 2000 was fine or you may drive this field using qs_len like
 mod_val = qs_len /1000